Client/Frontend/Sync/BraveCore/Bookmarkv2.swift

```diff
@@ -0,0 +1,316 @@
+// with the same layout/interface as `Bookmark (from CoreData)`
+class Bookmarkv2 {
+    private let bookmarkNode: BookmarkNode?
+    private static let bookmarksAPI = BraveBookmarksAPI()
+    
+    init(_ bookmarkNode: BookmarkNode) {
+        self.bookmarkNode = bookmarkNode
+    }
+    
+    public var isFolder: Bool {
+        return bookmarkNode?.isFolder == true
+    }
+    
+    public var title: String? {
+        return bookmarkNode?.titleUrlNodeTitle
+    }
+    
+    public var customTitle: String? {
+        return bookmarkNode?.titleUrlNodeTitle
+    }
+    
+    public var displayTitle: String? {
+        return self.customTitle
+    }
+    
+    public var url: String? {
+        return bookmarkNode?.titleUrlNodeUrl?.absoluteString
+    }
+    
+    public var domain: Domain? {
+        if let url = bookmarkNode?.url {
+            return Domain.getOrCreate(forUrl: url, persistent: true)
+        }
+        return nil
+    }
+    
+    public var created: Date? {
+        get {
+            return bookmarkNode?.dateAdded
+        }
+        
+        set {
+            bookmarkNode?.dateAdded = newValue ?? Date()
+        }
+    }
+    
+    public var parent: Bookmarkv2? {
+        get {
+            if let parent = bookmarkNode?.parent {
+                return Bookmarkv2(parent)
+            }
+            return nil
+        }
+    }
+    
+    public var children: [Bookmarkv2]? {
+        return bookmarkNode?.children.map({ Bookmarkv2($0) })
+    }
+    
+    public var canBeDeleted: Bool {
+        return bookmarkNode?.isPermanentNode == false
+    }
+    
+    public var objectID: Int {
+        return Int(bookmarkNode?.nodeId ?? 0)
+    }
+    
+    public var order: Int16 {
+        let children = bookmarkNode?.parent?.children
+        return Int16(children?.firstIndex(where: { $0.guid == self.bookmarkNode?.guid }) ?? -1)
+    }
+    
+    public func delete() {
+        if let bookmarkNode = bookmarkNode, self.canBeDeleted {
+            Bookmarkv2.bookmarksAPI.removeBookmark(bookmarkNode)
+        }
+    }
+}
+
+extension Bookmarkv2 {
+    
+    public class func addFolder(title: String, parentFolder: Bookmarkv2? = nil) {
+        if let parentFolder = parentFolder?.bookmarkNode {
+            Bookmarkv2.bookmarksAPI.createFolder(withParent: parentFolder, title: title)
+        } else {
+            Bookmarkv2.bookmarksAPI.createFolder(withTitle: title)
+        }
+    }
+    
+    public class func add(url: URL, title: String?, parentFolder: Bookmarkv2? = nil) {
+        if let parentFolder = parentFolder?.bookmarkNode {
+            Bookmarkv2.bookmarksAPI.createBookmark(withParent: parentFolder, title: title ?? "", with: url)
+        } else {
+            Bookmarkv2.bookmarksAPI.createBookmark(withTitle: title ?? "", url: url)
+        }
+    }
+    
+    public func existsInPersistentStore() -> Bool {
+        return true //Need a way to tell if a folder already exists in the sync chain..
```

Comment:
It is still needed. There is currently no way for me to know until it is synced. I'm trying to find a way to determine if it is even possible.

This is only a guard against one device having a folder and the other device not having that folder yet because it didn't sync over yet.

Atm, I call `reloadTable` so it doesn't have any effect. But once we switch that to be animated, we should figure out a way for this.